lib/Doctrine/Sniffs/Spacing/SpaceOnNotSniff.php

```diff
@@ -1,45 +1,89 @@
 <?php
-/*
-* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-* A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-* OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-* DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-* THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-*
-* This software consists of voluntary contributions made by many individuals
-* and is licensed under the MIT license. For more information, see
-* <http://www.doctrine-project.org>.
-*/
 
 declare(strict_types=1);
 
 namespace Doctrine\Sniffs\Spacing;
 
-use PHP_CodeSniffer\Sniffs\Sniff;
 use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
 
 final class SpaceOnNotSniff implements Sniff
 {
-    use EnsureSpaces;
-
     private const MESSAGE = 'There must be a single space %s a NOT operator; %d found';
 
-    public function register()
+    /**
+     * @return int[]
+     */
+    public function register() : array
     {
         return [\T_BOOLEAN_NOT];
     }
 
-    public function process(File $phpcsFile, $stackPtr)
+    /**
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
```

Comment:
I wish there was a way to not pollute code with standard-specific annotations and have exceptions described in an external file (like PHPStan does).

Reply:
I think this exact one (with no typehint) should be also suppressable by adding `@param` annotation.

Reply:
`phpcsSuppress` should be the exception - especially for this one. `MissingParameterTypeHint` requires either a type declaration on the parameter or a `@param` docblock, so there should be `@param`, not `@phpcsSuppress`.

Reply:
There isn't actually much cases where you would need that annotation, its use is mostly exceptional (like design errors i.e. write-only property).
Otherwise in this case `@param` is provided by parent declaration.

Reply:
Tried removing it, but realized it must be kept, because otherwise CS will report `int` not being used natively - but it can't be, parent forces mixed. Unfortunately CS has no way to know, it only has single file scope. :(